src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -177,12 +201,14 @@ def iot_dps_access_policy_update(cmd, client, dps_name, resource_group_name, acc
     if no_wait:
         return client.iot_dps_resource.begin_create_or_update(resource_group_name, dps_name, dps_description)
     LongRunningOperation(cmd.cli_ctx)(client.iot_dps_resource.begin_create_or_update(resource_group_name, dps_name, dps_description))
-    return iot_dps_access_policy_get(client, dps_name, resource_group_name, access_policy_name)
+    return iot_dps_policy_get(client, dps_name, access_policy_name, resource_group_name)
 
 
-def iot_dps_access_policy_delete(cmd, client, dps_name, resource_group_name, access_policy_name, no_wait=False):
+def iot_dps_policy_delete(cmd, client, dps_name, access_policy_name, resource_group_name=None, no_wait=False):
+    resource_group_name = _ensure_dps_resource_group_name(client, resource_group_name, dps_name)
     dps_access_policies = []
-    dps_access_policies.extend(iot_dps_access_policy_list(client, dps_name, resource_group_name))
+    dps_access_policies.extend(iot_dps_policy_list(client, dps_name, resource_group_name))
+
     if not _is_policy_existed(dps_access_policies, access_policy_name):
```

Comment:
I know this wasn't in the PR changes, but `_is_policy_existed` is not my favorite name. Can we change it to `_does_policy_exist` or similar?